[Keyboard] Add Dy16 with via compatibility (#9823)
qmk/qmk_firmware: keyboards/dreamystate/dy16/readme.md

# Dreamystate Dy16 Keyboard

> ![Dy16Photo](https://i.imgur.com/jpnMay9.jpeg)

Reviewer: Swaps this image for a smaller version.

```suggestion
![Dy16Photo](https://i.imgur.com/jpnMay9m.jpeg)``
```

> ![Dy16Photo](https://i.imgur.com/jpnMay9m.jpeg)``

Reviewer: I had a typo on my previous suggestion.

```suggestion
![Dy16Photo](https://i.imgur.com/jpnMay9m.jpeg)
```

> More: [Imgur](https://imgur.com/gallery/6tu9KdA).

Reviewer 2: ```suggestion

More: [Imgur](https://imgur.com/gallery/6tu9KdA).
```

> A 16 key macropad, with USB C, RGB light and Buzzer、 Encoder in the future.

Reviewer 2: ```suggestion
A 16 key macropad, with USB C, RGB light and Buzzer. Encoder in the future.
```

> # Dreamystate Dy16 Keyboard
> 
> A 16 key macropad, with USB C, RGB light and Buzzer、 Encoder in the future.
> 
> Keyboard Maintainer: Dreamystate
> Hardware Supported: The PCBs, Controllers supported  
> 
> ![Dy16Photo](https://i.imgur.com/jpnMay9.jpeg)
> More: [Imgur](https://imgur.com/gallery/6tu9KdA).
> 
> Make example for this keyboard (after setting up your build environment):
> 
> 	qmk compile -kb dreamystate/dy16 -km default
> 	or
> 	make dreamystate/dy16:default
> 
> See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) 
> and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. 
> Brand new to QMK? 
> Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
> 
> 

Reviewer 3: ```suggestion
# Dreamystate Dy16 Keyboard

![Dy16Photo](https://i.imgur.com/jpnMay9.jpeg)
More: [Imgur](https://imgur.com/gallery/6tu9KdA).

A 16 key macropad, with USB C, RGB light and Buzzer、 Encoder in the future.

* Keyboard Maintainer: Dreamystate
* Hardware Supported: Dy16 PCB 

Make example for this keyboard (after setting up your build environment):

    make dreamystate/dy16:default

Flashing example for this keyboard:

    make dreamystate/dy16:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
